src/test/java/io/strimzi/systemtest/DrainCleanerST.java

```diff
@@ -27,39 +28,60 @@ public class DrainCleanerST extends AbstractST {
 
     @Test
     void testEvictionRequestOnKafkaPod() {
-        final String stsName = "my-cluster-kafka";
-
-        LOGGER.info("Creating dummy pod that will contain \"kafka\" in its name.");
-        createStatefulSetAndPDBWithWait(stsName);
-
-        LOGGER.info("Creating eviction request to the pod");
-        String podName = kubeClient().listPodsByPrefixInName(Constants.NAMESPACE, stsName).get(0).getMetadata().getName();
-        kubeClient().getClient().pods().inNamespace(Constants.NAMESPACE).withName(podName).evict();
+        final String stsName = "my-cluster-foo";
+        final Map<String, String> labels = Map.of(
+                "app", stsName,
+                "strimzi.io/kind", "Kafka",
+                "strimzi.io/name", "my-cluster-kafka"
+        );
+
+        LOGGER.info("Creating dummy pod.");
+        createStatefulSetAndPDBWithWait(stsName, labels);
+
+        LOGGER.info("Creating eviction request to the pod.");
+        ObjectMeta meta = kubeClient().listPodsByPrefixInName(Constants.NAMESPACE, stsName).get(0).getMetadata();
+
+        kubeClient().getClient().pods().inNamespace(Constants.NAMESPACE).withName(meta.getName())
+            .evict(new EvictionBuilder()
+                .withMetadata(meta)
+                .withDeleteOptions(new DeleteOptionsBuilder()
+                    .withGracePeriodSeconds(0L)
+                    .build())
+                .build());
```

Comment:
Why was this changed? The original code should IMHO keep working with simple `evict()`.

Reply:
@scholzj looks like it doesn't work, at least on my env.

Looks like the ObjectMeta of the pod to be evicted is required, otherwise you don't get a match and this is what the webhook prints out (the 2nd should be a match):

```sh
2023-06-20 07:30:03,867 INFO  [io.str.ValidatingWebhook] (executor-thread-1) Received eviction event which does not match any relevant pods.
2023-06-20 07:31:16,672 INFO  [io.str.ValidatingWebhook] (executor-thread-0) Received eviction event which does not match any relevant pods.
```

Reply:
I'll do that.

Reply:
I get this in the real drain:

```
2023-06-20 09:45:32,976 INFO [io.str.ValidatingWebhook] (executor-thread-4) Received eviction event which does not match any relevant pods.
```

So I think the test needs to do the same it did before. And the code needs to get the pod first based on the name and then check the labels on the pod.

Reply:
Yeah, I did a test on my end and I can confirm.